lazyflow/request/threadPool.py

```diff
@@ -39,13 +40,32 @@ class ThreadPool:
     def __init__(self, num_workers: int):
         """Start all workers."""
         self.unassigned_tasks = queue.PriorityQueue()
+        self.ready_workers = queue.Queue()
 
-        self.workers = {_Worker(self, i) for i in range(num_workers)}
+        self.workers = [_Worker(self, i) for i in range(num_workers)]
         for w in self.workers:
             w.start()
 
         atexit.register(self.stop)
 
+        self._scheduler = threading.Thread(target=self._distribute_work)
+        self._scheduler.daemon = True
+        self._scheduler.start()
+
+    def _distribute_work(self):
+        while True:
+            try:
+                task = self.unassigned_tasks.get()
+                worker = getattr(task, "assigned_worker", None)
+
+                if not worker:
+                    worker = self.ready_workers.get()
+
+                worker.job_queue.put(task)
+                task = None
+            except Exception as e:
+                logger.exception("Exception in scheduler")
```

Comment:
```suggestion
                logger.exception(f"Unhandled exception in the scheduler thread: {e!r}")
```

Reply:
Kept the message except for the reference to exception object `{e!r}` as `logger.exception` already provides all exception information.

Reply:
That's great, didn't knew that!